pkg/lexer/lexer.go

```diff
@@ -191,15 +191,15 @@ func (l *Lexer) readIdent() string {
 
 func (l *Lexer) readString() (string, error) {
 	pos := l.pos
-	backslashCnt := 0
+	escaped := false
 	for {
 		if l.cur == '\\' {
```

Comment:
you could get "fancy" and just say `escaped = l.cur == '\\' && !escaped`. Perhaps that is a little too obtuse? (same in the JS too)

Reply:
yeah, i think even though this is a little "fancy" it is worth doing for the reduction in code size. updated.